orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/graph/StageGraphBuilder.java

```diff
@@ -132,6 +134,28 @@ public void connect(@Nonnull Stage previous, @Nonnull Stage next) {
     graph.putEdge(previous, next);
   }
 
+  /**
+   * Adds a new stage to the graph and makes it depend on the last stage that
+   * was added if any. This is convenient for straightforward stage graphs to
+   * avoid having to pass around references to stages in order to use
+   * {@link #connect(Stage, Consumer)}.
+   * <p>
+   * If no stages have been added so far, this is synonymous with calling
+   * {@link #add(Consumer)}.
+   *
+   * @param init See {@link #add(Consumer)}
+   * @return the newly created stage.
+   */
+  public @Nonnull Stage append(
+    @Nonnull Consumer<Stage> init
+  ) {
+    if (lastAdded == null) {
+      return add(init);
+    } else {
+      return connect(lastAdded, init);
+    }
+  }
```

Comment:
`append` is a great addition. Will update my other PR to use this instead.

Reply:
Yeah, I found myself doing exactly that logic in various stage builders. Now you only need to keep variables of stages if you want to do something more complex like a fork/join.